File: To_DO_List_Python/pages/home_page.py
from selenium.webdriver.common.by import By
from utilities.utility import Utility # type: ignore

class Home_Page:
    def __init__(self, chromeDriver):
        self.chromeDriver = chromeDriver

    # Locators
    InputField = (By.XPATH, "//input[@placeholder='Write your next task here...']")
    CheckBox = (By.XPATH, "//label[@title='Done']")
    TaskLabel = (By.XPATH, "//form[@class='edit_task']")
    DeleteButton = (By.XPATH, "//li[@class='delete']/a[@data-hint='Delete task']/i[@class='icon-remove']")

    # Actions
    def AddNewTask(self, Task):
        Utility.AddNewTask(self.chromeDriver, self.InputField, Task)
        return self

    def MarkAsCompleted(self):
        Utility.ClickOnElement(self.chromeDriver, self.CheckBox)
        return self

    def DeleteCompletedTask(self):
        Utility.hover_over_element(self.chromeDriver, self.TaskLabel)
        Utility.ClickOnElement(self.chromeDriver, self.DeleteButton)
        Utility.AcceptAlert(self.chromeDriver)

File: To_DO_List_Java/src/main/java/Pages/Home_Page.java
package Pages;

import Utilities.Utility;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class Home_Page {
WebDriver chromeDriver;
    //Constructor to make the two webDrivers in main and test is the same
    public Home_Page(WebDriver chromeDriver) {
        this.chromeDriver=chromeDriver;
    }

    //Locators
 private final By InputField = By.xpath("//input[@placeholder='Write your next task here...']");
 private final By CheckBox = By.xpath("//label[@title='Done']");
 private final By TaskLabel = By.xpath("//form[@class='edit_task']");
 private final By DeleteButton = By.xpath("//li[@class='delete']/a[@data-hint='Delete task']/i[@class='icon-remove']");

    //Actions
    //Adding new task
    public Home_Page AddNewTask(String Task){
        Utility.AddNewTask(chromeDriver,InputField,Task);
        return this;
    }

    //Mark the added task as completed
    public Home_Page MarkAsCompleted(){
        Utility.ClickOnElement(chromeDriver,CheckBox);
        return this;
    }

    //Delete the completed task
    public void DeleteCompletedTask(){
        Utility.hoverOverElement(chromeDriver,TaskLabel);
        Utility.ClickOnElement(chromeDriver,DeleteButton);
        Utility.AcceptAlert(chromeDriver);
    }

}
